File: Semestr1/IntroToComputerScience/lista_13/1.c
#include <stdio.h>
#include <stdbool.h>

typedef struct list
{
    struct list* next; 
    int val;
}list;

list insert(list lista, int val)
{
    list* root = lista;

}

bool** list_to_matrix(list* graph, int n)
{
    bool** temp = (bool**)malloc(n*sizeof(bool*));
    for(int i = 0; i < n; i++)
    {
        temp[i] = (bool*)malloc(n*sizeof(bool));
    }
    for(int i = 0; i < n; i++)
    {
        while(graph[i]->next != NULL)
        {
            temp[i][graph[i]->val] = true;
            temp[graph[i]->val][i] = true;
            graph[i] = graph[i]->next;
        }
        temp[i][graph[i]->val] = true;
        temp[graph[i]->val][i] = true;
    }
    return temp;
}

list* matrix_to_list(bool** matrix, int n)
{
    list* graph = (list*)malloc(n*sizeof(list));
    for(int i = 0; i < n; i++)
    {
        for(int j = 0; j < n; j++)
        {
            if(matrix[i][j] == true)
            {
                graph[i] = insert(graph[i], j);
            }
        }
    }
}

int main()
{

}

File: Semestr1/IntroToComputerScience/lista_13/1.py
class ListItem:
    def __init__(self, val):
        self.value = val
        self.next = None

def insert_end(list, value):
    while list.next != None:
        list = list.next
    list.next = ListItem(value)
def print_list(list):
    while list.next != None:
        print(list.val, end=" ")
        list = list.next
    print(list.val)
    
def matrix_to_list(matrix, list, n):
    for i in range(n):
        root = list[i]
        for j in range(n):
            if matrix[i][j] == 1:
                list[i] = insert_end(list[i], j)
        list[i] = root
    return list

def list_to_matrix(matrix, list, n):
    for i in range(n):
        while list[i].next != None:
            matrix[i][list[i].val] = 1
            matrix[list[i].val][i] = 1
            list[i] = list[i].next
        matrix[i][list[i].val] = 1
        matrix[list[i].val][i] = 1
        

            